modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js

```diff
@@ -181,7 +181,7 @@ const doEvaluate = debounce((fieldName, evaluatorContext, callback) => {
 			callback(null, mergedPages);
 		})
 		.catch((error) => callback(error));
-}, 600);
+}, 200);
```

Comment:
we are playing timing games here. in the previous solution added a time out, but the bug persisted if the user is faster enough, now we are reducing delay. we should seek a different solution. lets work on this together.

Reply:
Perfect! Thanks for the heads-up! I'll reach you on Slack so we can schedule a meeting. In the meantime, I'll be working on finding a different approach.